### PR TITLE
Fix #2775 - running pytest with "--pyargs" will result in Items with …

### LINKED ISSUE #2775: Running pytest with "--pyargs" will result in Items with empty "parent.nodeid" if run from a different root directory
When running pytest with "--pyargs" from a directory that isn't a parent directory of the packages under test, then the Items created will have an empty parent.nodeid (i.e. "::test_function").

This is because the _makeid() method in FSCollector compares the path of the Module to the rootdir:
```
relpath = self.fspath.relto(self.config.rootdir)
```

### PR DESCRIPTION
**Pull request by cryporchild**
…empty "parent.nodeid" if run from a different root directory

### CODE REVIEW

**Comment by nicoddemus** [comment]
xref: #2775

File: _pytest/main.py [hunk 1]
```diff
@@ -520,8 +520,18 @@ def __init__(self, fspath, parent=None, config=None, session=None):
         super(FSCollector, self).__init__(name, parent, config, session)
         self.fspath = fspath
 
+    def _check_initialpaths_for_relpath(self):
+        for initialpath in self.session._initialpaths:
+            parent_path = self.fspath
+            for _ in parent_path.parts():
```
Comment (RonnyPfannschmidt) [inline _pytest/main.py]: This looks just wrong

Comment (cryporchild) [reply _pytest/main.py]: Yep, that's a fair comment. I've updated this to (hopefully) use the py.path apis properly.

**Comment by nicoddemus** [comment]
@cryporchild gentle ping.

**Comment by nicoddemus** [comment]
Thanks @cryporchild, and sorry for this taking so long! 👍

**Pull request merged**